includes/class-newspack-newsletters.php

```diff
@@ -1057,6 +1084,32 @@ public static function debug_mode() {
 		return defined( 'NEWSPACK_NEWSLETTERS_DEBUG_MODE' ) ? NEWSPACK_NEWSLETTERS_DEBUG_MODE : false;
 	}
 
+
+	/**
+	 * Get properties required to render a useful modal in the editor that alerts
+	 * users of ads they're sending.
+	 *
+	 * @param WP_REST_REQUEST $request The WP Request Object.
+	 * @return array
+	 */
+	public static function get_ads_warning_in_editor( $request ) {
+		$letterhead                 = new Newspack_Newsletters_Letterhead();
+		$has_letterhead_credentials = $letterhead->has_api_credentials();
+		$post_date                  = $request->get_param( 'date' );
+		$newspack_ad_type           = Newspack_Newsletters_Ads::NEWSPACK_NEWSLETTERS_ADS_CPT;
+		$url_to_manage_promotions   = 'https://app.tryletterhead.com/promotions';
+		$url_to_manage_newspack_ads = "/wp-admin/edit.php?post_type={$newspack_ad_type}&page=newspack-newsletters-ads-admin";
+
+		$ads = Newspack_Newsletters_Renderer::get_ads( $post_date, 0 );
+
+		return [
+			'count'     => count( $ads ),
+			'label'     => $has_letterhead_credentials ? 'promotion' : 'ad',
```

Comment:
These two strings should be translatable (wrapped in `__()`). Ideally, the label for the ads modal should be returned from the backend, so there's no need for the `adLabel === 'promotion'` logic.

Reply:
Good idea.